middleware.go

```diff
@@ -14,155 +11,63 @@ import (
 	"github.com/square/go-jose/v3/jwt"
 )
 
-//ContextKey used to access values in the context
-type ContextKey string
-
-const (
-	//ContextJWTError is used to access the JWT error in the context
-	ContextJWTError ContextKey = "JWT_ERROR"
-	//ContextJWTObject is used to access the JWT object in the context
-	ContextJWTObject ContextKey = "JWT_OBJECT"
-	//ContextJWTClaims is used to access the JWT claims in the context
-	ContextJWTClaims ContextKey = "JWT_CLAIMS"
-	//ContextValidatorError is used to access the error provided by validators
-	ContextValidatorError ContextKey = "VALIDATOR_ERROR"
-
-	//ErrInvalidToken is returned as a error if the token was invalid
-	ErrInvalidToken = "The token provided was invalid"
-)
-
-//JWTMiddleware is used to mange JWT middleware
-type JWTMiddleware struct {
-	JWKS        *jose.JSONWebKeySet
-	Issuer      string
-	EnableDebug bool
-}
+type JWTResolver func(w http.ResponseWriter, r *http.Request) (JWTInfo, error)
 
-//NewJWTMiddleware is used to greate a new JWTMiddleware
-func NewJWTMiddleware(jwks *jose.JSONWebKeySet, issuer string) JWTMiddleware {
-	return JWTMiddleware{
-		JWKS:        jwks,
-		Issuer:      issuer,
-		EnableDebug: false,
-	}
+type JWTInfo struct {
```

Comment:
Codacy found an issue: [exported type JWTInfo should have comment or be unexported](https://app.codacy.com/manual/johnfg10/gojwt/pullRequest?prid=5899569)